### PR TITLE
feat: upgrade llamacloud index

### PR DESCRIPTION
**Pull request by EmanuelCampos**

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "pgvector": "^0.1.8",
     "portkey-ai": "^0.1.16",
     "rake-modified": "^1.0.8",
+    "replicate": "^0.30.2",
```
Comment (Disiok) [inline packages/core/package.json:43]: why are we bringing in replicate?

File: examples/cloud/query.ts [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
 async function main() {
   const index = new LlamaCloudIndex({
     name: "test",
     projectName: "default",
-    baseUrl: process.env.LLAMA_CLOUD_BASE_URL,
-    apiKey: process.env.LLAMA_CLOUD_API_KEY,
-  });
-  const queryEngine = index.asQueryEngine({
-    denseSimilarityTopK: 5,
+    apiKey: "llx-9xAW94gy4y2jj7iJhj0nlcYZvv0thTDDjkKgrjBW7LyO5Ygn",
```
Comment (marcusschiesser) [inline examples/cloud/query.ts:9]: @EmanuelCampos oops

File: packages/core/src/cloud/utils.ts [hunk 3]
```diff
@@ -11,18 +12,34 @@ export function getAppBaseUrl(baseUrl?: string): string {
...
   apiKey,
   baseUrl,
-}: ClientParams = {}): Promise<PlatformApiClient> {
+}: ClientParams = {}): typeof LlamaCloud {
   // Get the environment variables or use defaults
   baseUrl = getBaseUrl(baseUrl);
   apiKey = apiKey ?? getEnv("LLAMA_CLOUD_API_KEY");
 
-  const { PlatformApiClient } = await import("@llamaindex/cloud");
+  if (!apiKey) {
```
Comment (marcusschiesser) [inline packages/core/src/cloud/utils.ts:23]: i remember @llamaindex/cloud didn't work with CJS, that's why there's the dynamic import - is this fixed?

File: packages/core/src/ingestion/IngestionPipeline.ts [hunk 4]
```diff
@@ -129,50 +127,4 @@ export class IngestionPipeline {
...
   }
-
-  private async getClient(): Promise<PlatformApiClient> {
-    if (!this.client) {
-      this.client = await getClient(this.clientParams);
-    }
-    return this.client;
-  }
-
-  async register(params: {
```
Comment (marcusschiesser) [inline packages/core/src/ingestion/IngestionPipeline.ts:140]: no register anymore?

File: packages/core/src/cloud/config.ts [hunk 5]
```diff
@@ -16,7 +16,7 @@ function getTransformationConfig(
 ): PlatformApi.ConfiguredTransformationItem {
   if (transformation instanceof SimpleNodeParser) {
     return {
-      configurableTransformationType: "SENTENCE_AWARE_NODE_PARSER",
+      configurable_transformation_type: "SENTENCE_AWARE_NODE_PARSER",
```
Comment (marcusschiesser) [inline packages/core/src/cloud/config.ts:19]: didn't fern have a feature to support camel case for TS?

File: packages/core/src/cloud/LlamaCloudIndex.ts [hunk 6]
```diff
@@ -69,94 +76,74 @@ export class LlamaCloudIndex {
...
+    });
 
-      for await (const execution of executionsIds) {
-        const dataSourceExecution =
-          await client.dataSource.getDataSourceExecution(
-            execution.dataSourceId,
-            execution.exectionId,
+    while (true) {
+      const allDocumentsStatus = await Promise.all(
+        params.documents.map(async (doc) => {
```
Comment (marcusschiesser) [inline packages/core/src/cloud/LlamaCloudIndex.ts:103]: @EmanuelCampos we have a concurrency limiter now, see LlamaParse reader code

File: packages/core/src/cloud/LlamaCloudRetriever.ts [hunk 7]
```diff
@@ -55,26 +68,32 @@ export class LlamaCloudRetriever implements BaseRetriever {
...
-    ).pipeline.runSearch(pipelines[0].id, {
-      ...this.retrieveParams,
-      query,
-      searchFilters: preFilters as Record<string, unknown[]>,
-    });
 
-    const nodes = this.resultNodesToNodeWithScore(results.retrievalNodes);
+    console.log("Retrieving with pipeline", pipeline.id, "for query", query);
+
+    console.log("retrieveParams", this.retrieveParams);
```
Comment (himself65) [inline packages/core/src/cloud/LlamaCloudRetriever.ts:84]: remove

File: packages/core/src/cloud/config.ts (hunk 5 above)
Comment (himself65) [reply packages/core/src/cloud/config.ts:19]: I don't use fern in TS side for the latest code, but it's okay to use any type generation in SDK side

**Review by himself65 — CHANGES_REQUESTED** [review]

File: packages/core/src/cloud/LlamaCloudRetriever.ts [hunk 8]
```diff
@@ -1,28 +1,37 @@
...
+  ClientParams,
+  CloudConstructorParams,
+  ObjectToCamel,
+} from "./types.js";
 import { DEFAULT_PROJECT_NAME } from "./types.js";
 import { getClient } from "./utils.js";
+
 export type CloudRetrieveParams = Omit<
-  PlatformApi.RetrievalParams,
-  "query" | "searchFilters" | "pipelineId" | "className"
```
Comment (himself65) [inline packages/core/src/cloud/LlamaCloudRetriever.ts:12]: This sounds not right, it's safe in type but not in runtime, you have to convert it to camel case

**Review by himself65 — CHANGES_REQUESTED** [review]

File: packages/core/src/ingestion/IngestionPipeline.ts (hunk 4 above)
Comment (EmanuelCampos) [reply packages/core/src/ingestion/IngestionPipeline.ts:140]: yup! we are getting rid of this

File: packages/core/src/cloud/utils.ts (hunk 3 above)
Comment (EmanuelCampos) [reply packages/core/src/cloud/utils.ts:23]: hmm will check this

File: examples/cloud/query.ts (hunk 2 above)
Comment (Disiok) [reply examples/cloud/query.ts:9]: @EmanuelCampos plz revoke this key

Comment (EmanuelCampos) [reply examples/cloud/query.ts]: is a development one

**Pull request merged**